src/allocator.rs

```diff
@@ -4,8 +4,52 @@ use crate::reduction::EvalErr;
 use chia_bls::{G1Element, G2Element};
 use clvm_traits::{ClvmDecoder, ClvmEncoder, FromClvmError, ToClvmError};
 
+const MAX_NUM_ATOMS: usize = 62500000;
+const MAX_NUM_PAIRS: usize = 62500000;
+const NODE_PTR_IDX_BITS: u32 = 26;
+const NODE_PTR_IDX_MASK: u32 = (1 << NODE_PTR_IDX_BITS) - 1;
+
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct NodePtr(pub i32);
+pub struct NodePtr(pub u32);
```

Comment:
It would be nice if we can make this private, so that the inner type representation can be changed without breaking semver compatibility. Is there anything currently that relies on this value?

Reply:
good point. my intention of implementing `Default` was for nothing to depend on this. but it seems there still are a few places. I'll investigate

Reply:
sadly, there's one ugly hack left. for now I just called it out as a hack to be fixed in the future